cpp/src/arrow/builder.h

```diff
@@ -143,8 +143,6 @@ class ARROW_EXPORT PrimitiveBuilder : public ArrayBuilder {
   explicit PrimitiveBuilder(MemoryPool* pool, const TypePtr& type)
       : ArrayBuilder(pool, type), data_(nullptr) {}
 
-  virtual ~PrimitiveBuilder() {}
```

Comment:
Seems like there is some intention here for deleting the virtual destructors? What's the reasoning for this?